app/server/models/transfer_account.py

```diff
@@ -125,34 +128,92 @@ def set_balance_offset(self, val):
         self.update_balance()
 
     def update_balance(self):
+        """
+        Update the balance of the user by calculating the difference between inbound and outbound transfers, plus an
+        offset.
+        For inbound transfers we count ONLY complete, while for outbound we count both COMPLETE and PENDING.
+        This means that users can't spend funds that are potentially:
+        - already spent
+        or
+        - from a transfer that may ultimately be rejected.
+        """
         if not self._balance_offset_wei:
             self._balance_offset_wei = 0
-        net_credit_transfer_position_wei = (self.total_received - self.total_sent) * int(1e16)
+
+        net_credit_transfer_position_wei = (
+                self.total_received_complete_only_wei - self.total_sent_incl_pending_wei
+        )
+
         self._balance_wei = net_credit_transfer_position_wei + self._balance_offset_wei
 
     @hybrid_property
-    def total_sent(self):
-        amount_cents = (
+    def total_sent_complete_only_wei(self):
+        """
+        The total sent by an account, counting ONLY transfers that have been resolved as complete locally
+        """
+        amount = (
+            db.session.query(
+                func.sum(server.models.credit_transfer.CreditTransfer._transfer_amount_wei).label('total')
+            )
+                .execution_options(show_all=True)
+                .filter(server.models.credit_transfer.CreditTransfer.sender_transfer_account_id == self.id)
+                .filter(server.models.credit_transfer.CreditTransfer.transfer_status == TransferStatusEnum.COMPLETE)
+                .first().total
+        )
+        return amount or 0
+
+    @hybrid_property
+    def total_received_complete_only_wei(self):
+        """
+        The total received by an account, counting ONLY transfers that have been resolved as complete
+        """
+        amount = (
+            db.session.query(
+                func.sum(server.models.credit_transfer.CreditTransfer._transfer_amount_wei).label('total')
+            )
+                .execution_options(show_all=True)
+                .filter(server.models.credit_transfer.CreditTransfer.recipient_transfer_account_id == self.id)
+                .filter(server.models.credit_transfer.CreditTransfer.transfer_status == TransferStatusEnum.COMPLETE)
+                .first().total
+        )
+
+        return amount or 0
```

Comment:
I think it would make sense to add unit tests for `total_received_complete_only_wei` and `total_sent_complete_only_wei`. Currently only the latter is covered.